docs/querying/api.md

```diff
@@ -433,13 +433,35 @@ URL query parameters:
   series from which to read the label values. Optional.
 - `limit=<number>`: Maximum number of returned series. Optional. 0 means disabled.
 
-
 The `data` section of the JSON response is a list of string label values.
 
-This example queries for all label values for the `job` label:
+This example queries for all label values for the `job_name` label:
+
+```json
+$ curl http://localhost:9090/api/v1/label/job_name/values
+{
+   "status" : "success",
+   "data" : [
+      "node",
+      "prometheus"
+   ]
+}
+```
+
+For label names that contain characters that are not valid in the legacy
+Prometheus character set (letters, numbers, underscores, and colons), they
```

Comment:
Strictly speaking, you can use non-legacy characters here as long as they are valid as URL components, can't we?

So I guess `http://localhost:9090/api/v1/label/job.name/values` would actually work?

If that's true, we only need the `U__` encoding for characters that don't work as URL components (in particular `/`).

Reply:
I was thinking about that, but that might lead people to use HTML encoding: "job%20name" which is not supported. So while job.name would work, I'd prefer to make the reasoning simpler and just ask that people use encoding when the name is non-legacy-compatible.

Reply:
By HTML encoding, you mean percent encoding, also known as URL encoding, don't you? I.e. https://en.wikipedia.org/wiki/Percent-encoding

I would expect that URL encoding is in fact supported, because that should just be dealt with by the Go HTTP libraries and the Prometheus code wouldn't even notice.

It might very well possible that we only need the `U__` encoding for a `/` in the label name. (At least that was the case long ago for label _values_ as they show up in URLs for pushing to the PGW. Of course, back then there was no `U__` encoding, so we elected for base64.)

Sorry for beating this point for so long, but it feels confusing to refer to the legacy Prometheus character set here when there is no technical reason for it.

Reply:
I think I have rewritten it this way.